### PR TITLE
Create users if not found.

### PR DESCRIPTION
**Pull request by bkyryliuk**
This change will allow to pre-create and preassign users to the groups before they login to the superset. It will allow users to have full access once they are logged in to the tool.

Reviewers:
* @vera-liu 
* @mistercrunch 
* @ascott

### CODE REVIEW

File: superset/views.py [hunk 1]
```diff
@@ -1139,16 +1139,33 @@ class Superset(BaseSupersetView):
...
+        created_users = []
+        granted_users = []
+        for user_data in users_data:
+            user = sm.find_user(username=user_data['username'])
+            if not user and user_data['username']:
+                sm.add_user(
+                    username=user_data['username'],
+                    first_name=user_data['username'],
+                    last_name=user_data['username'],
+                    email=user_data['username'],
```
Comment (ascott) [inline superset/views.py]: why are we using `username` as email?

File: superset/views.py [hunk 2]
```diff
@@ -1139,16 +1139,33 @@ class Superset(BaseSupersetView):
...
-            user = sm.find_user(username=username)
-            if user:
+        created_users = []
+        granted_users = []
+        for user_data in users_data:
+            user = sm.find_user(username=user_data['username'])
+            if not user and user_data['username']:
+                sm.add_user(
+                    username=user_data['username'],
+                    first_name=user_data['username'],
```
Comment (ascott) [inline superset/views.py]: does `user_data` have a first/last name available?

File: superset/views.py (hunk 1 above)
Comment (bkyryliuk) [inline superset/views.py]: typo :) will improve the unit test

File: tests/access_tests.py [hunk 3]
```diff
@@ -370,7 +373,10 @@ def test_update_role(self):
         resp = self.client.post(
             '/superset/update_role/',
             data=json.dumps({
-                'usernames': ['gamma'],
+                'users': [{
+                        'username': 'gamma',
+                        'email': 'gamma@superset.com'
```
Comment (bkyryliuk) [inline tests/access_tests.py]: I've added the email field, actually we plan to send emails for the access requests )

**Comment by ascott** [comment]
probably want @mistercrunch do take a quick look, but LGTM!

**Comment by mistercrunch** [comment]
LGTM

**Pull request merged**